File: api/src/main/java/com/ruby/rubia_server/core/entity/ConversationMedia.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.base.BaseEntity;
import com.ruby.rubia_server.core.enums.MediaType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Table(name = "conversation_media")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ConversationMedia implements BaseEntity {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company; // Para multi-tenancy

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id", nullable = false)
    private Conversation conversation; // A conversa à qual esta mídia pertence

    @Column(name = "file_url", nullable = false)
    private String fileUrl; // A URL/caminho para o arquivo armazenado (ex: no S3, GCS)

    @Enumerated(EnumType.STRING)
    @Column(name = "media_type", nullable = false)
    private MediaType mediaType; // Tipo de mídia (ex: IMAGE, VIDEO, AUDIO, DOCUMENT)

    @Column(name = "mime_type", length = 100)
    private String mimeType; // Tipo MIME do arquivo (ex: "image/jpeg", "application/pdf")

    @Column(name = "original_file_name", length = 255)
    private String originalFileName; // Nome original do arquivo antes de ser salvo no storage

    @Column(name = "file_size_bytes")
    private Long fileSizeBytes; // Tamanho do arquivo em bytes

    @Column(name = "checksum", length = 64) // Opcional: para verificação de integridade (ex: SHA-256)
    private String checksum;

    @CreationTimestamp
    @Column(name = "uploaded_at", nullable = false, updatable = false)
    private LocalDateTime uploadedAt; // Quando a mídia foi armazenada/registrada

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    // Opcional: Quem enviou/recebeu este arquivo originalmente
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by_user_id")
    private User uploadedByUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by_customer_id")
    private Customer uploadedByCustomer;
}

File: api-dotnet/Rubia.Server/Entities/ConversationMedia.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("conversation_media")]
public class ConversationMedia : BaseEntity
{
    [Required]
    [Column("file_url")]
    public string FileUrl { get; set; } = string.Empty; // A URL/caminho para o arquivo armazenado

    [Column("media_type")]
    [Required]
    public MediaType MediaType { get; set; } // Tipo de mídia (ex: IMAGE, VIDEO, AUDIO, DOCUMENT)

    [Column("mime_type")]
    [MaxLength(100)]
    public string? MimeType { get; set; } // Tipo MIME do arquivo

    [Column("original_file_name")]
    [MaxLength(255)]
    public string? OriginalFileName { get; set; } // Nome original do arquivo

    [Column("file_size_bytes")]
    public long? FileSizeBytes { get; set; } // Tamanho do arquivo em bytes

    [Column("checksum")]
    [MaxLength(64)]
    public string? Checksum { get; set; } // Para verificação de integridade (ex: SHA-256)

    [Column("uploaded_at")]
    [Required]
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow; // Quando a mídia foi armazenada/registrada

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    [Column("conversation_id")]
    [Required]
    public Guid ConversationId { get; set; }
    
    [ForeignKey("ConversationId")]
    public virtual Conversation Conversation { get; set; } = null!;

    [Column("uploaded_by_user_id")]
    public Guid? UploadedByUserId { get; set; }
    
    [ForeignKey("UploadedByUserId")]
    public virtual User? UploadedByUser { get; set; }

    [Column("uploaded_by_customer_id")]
    public Guid? UploadedByCustomerId { get; set; }
    
    [ForeignKey("UploadedByCustomerId")]
    public virtual Customer? UploadedByCustomer { get; set; }

    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
}